roles/test-utils/sv1-mining-device/src/client.rs

```diff
@@ -69,8 +67,8 @@ impl Client {
     ///    the information from `sender_share`, it is formatted as a `v1::client_to_server::Submit`
     ///    and then serialized into a json message that is sent to the Upstream via
     ///    `sender_outgoing`.
-    pub(crate) async fn connect(client_id: u32) {
-        let stream = std::sync::Arc::new(TcpStream::connect(ADDR).await.unwrap());
+    pub async fn connect(client_id: u32, upstream_addr: std::net::SocketAddr) {
```

Comment:
```suggestion
    pub async fn connect(client_id: u32, upstream_addr: SocketAddr) {
```
`std::net::SocketAddr` can be added to the import statement above.